lib/video_player_web_hls.dart

```diff
@@ -163,41 +164,40 @@ class _VideoPlayer {
     // ignore: undefined_prefixed_name
     ui.platformViewRegistry.registerViewFactory(
         'videoPlayer-$textureId', (int viewId) => videoElement);
-    if(uri.toString().contains("m3u8"))
-    {
+    if (isSupported() && uri.toString().contains("m3u8")) {
       try {
         Hls hls = new Hls();
         hls.attachMedia(videoElement);
 
         hls.on('hlsMediaAttached', allowInterop((_, __) {
           hls.loadSource(uri.toString());
         }));
-        hls.on('hlsError', allowInterop((_,dynamic data){
+        hls.on('hlsError', allowInterop((_, dynamic data) {
           eventController.addError(PlatformException(
             code: _kErrorValueToErrorName[2],
             message: _kDefaultErrorMessage,
             details: _kErrorValueToErrorDescription[5],
           ));
-
-
         }));
 
+        videoElement.onCanPlay.listen((dynamic _) {
+          if (!isInitialized) {
+            isInitialized = true;
+            sendInitialized();
+          }
+        });
+      } catch (e) {
+        throw NoScriptTagException();
       }
-      catch(e)
-    {
-      throw NoScriptTagException();
-
-    }
+    } else {
+      videoElement.src = uri.toString();
+      videoElement.addEventListener('loadedmetadata', (_) {
+        if (!isInitialized) {
+          isInitialized = true;
+          sendInitialized();
+        }
+      });
```

Comment:
This is the main change. `onCanPlay` is only listened to if HLS is used, and when native video is used, we listen for 'metadataloaded'.

This follows the guidelines set in the hls.js readme.